forge-gui/res/cardsfolder/n/nahiri_the_lithomancer.txt

```diff
@@ -3,9 +3,10 @@ ManaCost:3 W W
 Types:Legendary Planeswalker Nahiri
 Loyalty:3
 Text:CARDNAME can be your commander.
-A:AB$ Token | Cost$ AddCounter<2/LOYALTY> | Planeswalker$ True | TokenAmount$ 1 | TokenScript$ w_1_1_kor_soldier | TokenOwner$ You | RememberTokens$ True | SubAbility$ DBAttach | SpellDescription$ Create a 1/1 white Kor Soldier creature token. You may attach an Equipment you control to it.
-SVar:DBAttach:DB$ Attach | Optional$ True | Choices$ Equipment.YouCtrl | ChoiceTitle$ Choose an Equipment you control | Defined$ Remembered | SubAbility$ DBCleanup | StackDescription$ None
-SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+A:AB$ Token | Cost$ AddCounter<2/LOYALTY> | Planeswalker$ True | TokenScript$ w_1_1_kor_warrior | RememberTokens$ True | SubAbility$ DBChooseToken | SpellDescription$ Create a 1/1 white Kor Warrior creature token. You may attach an Equipment you control to it.
+SVar:DBChooseToken:DB$ ChooseCard | DefinedCards$ Remembered | Mandatory$ True | ChoiceTitle$ Choose a token | SubAbility$ DBAttach | StackDescription$ None
```

Comment:
what does the corner case look like?
this seems to have worked already before

Reply:
I think it was in the case of Doubling Season and similar, the player should have ability to choose which token gets an attachment. @Hanmac may remember better

Not sure why we didn't bring Nahiri, Heir of Ancients and this to parity before 🤷‍♂️ they definitely should be the same one way or the other

Reply:
yea and when I tested that with the old version it already lets you choose the token

Reply:
Imo it would be better to have the choice inside the Attach

Reply:
There is my thoughts about refactoring 
https://github.com/Card-Forge/forge/issues/2289#issuecomment-1399565687